Network/Socket/SockAddr.hs

```diff
@@ -41,7 +45,27 @@ connect = G.connect
 -- 'defaultPort' is passed then the system assigns the next available
 -- use port.
 bind :: Socket -> SockAddr -> IO ()
-bind = G.bind
+bind s a = case a of
+  SockAddrUnix p -> do
+    -- gracefully handle the fact that UNIX systems don't clean up closed UNIX
+    -- domain sockets, inspired by https://stackoverflow.com/a/13719866
+    res <- try (G.bind s a)
+    case res of
+      Right () -> pure ()
```

Comment:
We don't use `pure` at this moment. Please use `return` instead.

Reply:
Done in 8d5ff8d